lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -4028,6 +4028,33 @@ CheckedCastKind TypeChecker::typeCheckCheckedCast(Type fromType,
     if (toType->isAnyObject() || fromType->isAnyObject())
       return CheckedCastKind::ValueCast;
 
+    // A cast from an existential type to a concrete type does not succeed.
+    //
+    // struct S {}
+    // enum FooError: Error { case bar }
+    // func foo() throws {
+    //   do {
+    //     throw FooError.bar
+    //   } catch is X {
+    //     print("Caught bar error")
+    //   }
+    // }
+    if (fromExistential) {
+      if (auto NTD = toType->getAnyNominal()) {
+        if (NTD->getAllConformances().empty()) {
```

Comment:
This is probably doing more work than it's saving! I think the other part is sufficient.

Reply:
@jrose-apple Okay, I have removed the call to `getAllConformances()`!